pkg/operators/ebpf/tracer.go

```diff
@@ -170,7 +175,11 @@ func (t *Tracer) receiveEventsFromPerfReader(gadgetCtx operators.GadgetContext)
 	for {
 		rec, err := t.perfReader.Read()
 		if err != nil {
-			return err
+			if errors.Is(err, perf.ErrClosed) {
+				return nil
+			}
+			gadgetCtx.Logger().Warnf("error reading perf event: %v", err)
```

Comment:
How often/when does this actually happen? Maybe we should just increase the drop counter instead of sending a warning?

Reply:
I don't expect it to happen to too often otherwise we would have cases where gadgets should have stopped working. But I might be wrong. @alban do you have any thoughts?

Reply:
This specific PR is not about lost events because ig can't keep up, but about a bug when the gadget generates bad events (see details in https://github.com/cilium/ebpf/issues/1633). So I think the patch is good.